dev-client/src/hooks/appPermissionsHooks.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useContext} from 'react';
+
+import {PermissionResponse} from 'expo-location';
+
+import {ForegroundPermissionsContext} from 'terraso-mobile-client/context/AppPermissionsContext';
+
+// As of Sept 2024, the app permissions hooks supplied by the expo libraries don't trigger component updates
+// when app permissions are updated. Instead, the permissions object stays stale until the next call to
+// get() or request().
+// This hook wraps the library-supplied hooks and should cause components to update when permissions are
+// updated in the app or after they are updated outside the app.
+export type UpdatedPermissionsHookReturnType = [
+  PermissionResponse | null, // permission
+  () => Promise<PermissionResponse>, // request
+  () => Promise<PermissionResponse>, // get
+];
+export type UpdatedPermissionsHookType = () => UpdatedPermissionsHookReturnType;
```

Comment:
Soliciting advice: The types feel a little sloppy between here and in AppPermissionsContext.tsx. 
FYI
- `UpdatedPermissionsHookReturnType` aims to match types compatible with `PermissionsRequestWrapper` (which matches the expo permission hooks' types)
- `ForegroundPermissionsType` is similar, but allows additional nulls for get and request because I wanted something easy to set up the context with. 
- The context provider should populate with actual values when it is mounted on app launch, so the hook expects them not to be null.

I’m also torn about if `ForegroundPermissionsType` should deviate further from `UpdatedPermissionsHookReturnType`, since it would be nice to have an object with named keys, but the array matches the way everything else is structured.

Reply:
thoughts:
- rather than initializing with `[null, null, null]`, you could initialize with `null`, and  `ForegroundPermissionsType` could become `UpdatedPermissionsHookReturnType | null`, or `UpdatedPermissionsHookReturnType` could become `NonNullable<ForegroundPermissionsType>`
- i agree about it seeming nicer to have named keys, i understand why react does this with `useState` but don't like how it seems to get cargo culted onto everyone's hook definitions when it doesn't make as much sense